remote/remote.go

```diff
@@ -75,13 +75,14 @@ func (rc remoteConfigProvider) WatchChannel(rp viper.RemoteProvider) (<-chan *vi
 func getConfigManager(rp viper.RemoteProvider) (crypt.ConfigManager, error) {
 	var cm crypt.ConfigManager
 	var err error
+	var kr *os.File
 
 	if rp.SecretKeyring() != "" {
-		kr, err := os.Open(rp.SecretKeyring())
-		defer kr.Close()
+		kr, err = os.Open(rp.SecretKeyring())
```

Comment:
```suggestion
		var kr *os.File
		kr, err = os.Open(rp.SecretKeyring())
```

Reply:
@sagikazarmark If I read your suggestion correctly, the only change was to move the declaration of the `kr` variable, which I've done and squashed into the original commit.

Reply:
@alrs that's correct, thank you!